zk-token-sdk/src/encryption/elgamal.rs

```diff
@@ -155,12 +155,16 @@ impl ElGamal {
 #[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize, Zeroize)]
 pub struct ElGamalKeypair {
     /// The public half of this keypair.
-    pub public: ElGamalPubkey,
+    public: ElGamalPubkey,
     /// The secret half of this keypair.
-    pub secret: ElGamalSecretKey,
+    secret: ElGamalSecretKey,
 }
 
 impl ElGamalKeypair {
+    pub fn new(public: ElGamalPubkey, secret: ElGamalSecretKey) -> Self {
```

Comment:
Does this method have a use case outside of tests?

Reply:
I don't imagine there should be much use. This is mainly used in tests to combine invalid combinations of `ElGamalPubkey` and `ElGamalSecretKey` to create a wrong `ElGamalKeypair`.

Reply:
Ah perhaps it is a good idea to put this function under a feature for tests.

Reply:
Yeah, a `#[cfg(test)]` attribute would be ideal. Unfortunately, that probably will not work, since it's needed in the `/programs` tests and benches.
How about renaming to `new_for_tests()` for now? (We may be able to make use of [this work](https://github.com/solana-labs/solana/pull/32169) later)

Reply:
Okay makes sense! Renamed the function to `new_for_tests`.